### PR TITLE
ops/model.py: Add Model.name

### PR DESCRIPTION
**Pull request by jameinel**
main.py now reads JUJU_MODEL_NAME to be able to set Model.name.

fixes: #274

### CODE REVIEW

File: ops/model.py [hunk 1]
```diff
@@ -47,7 +49,9 @@ class Model:
             for Kubernetes charms.
     """
 
-    def __init__(self, unit_name, meta, backend):
+    def __init__(
+            self, unit_name: str, meta: 'ops.charm.CharmMeta',
+            backend: '_ModelBackend', model_name: str = None):
```
Comment (chipaca) [inline ops/model.py:54]: is it time to require keyword arguments?

File: ops/main.py [hunk 2]
```diff
@@ -290,7 +290,8 @@ def main(charm_class):
     metadata, actions_metadata = _load_metadata(charm_dir)
     meta = ops.charm.CharmMeta(metadata, actions_metadata)
     unit_name = os.environ['JUJU_UNIT_NAME']
-    model = ops.model.Model(unit_name, meta, model_backend)
+    model_name = os.environ.get('JUJU_MODEL_NAME', None)
```
[the remark this answers is not in the document]
Comment (jameinel) [reply ops/main.py:293]: I realize it is the default, I find it conveys "I'm expecting this to be None if it isn't set". It is fair to say that is what you are doing by using get(). But the fact that you aren't sure that I'm aware is the point that I'm trying to convey to the next person who may or may not be aware.

If we prefer not to have it, I'll happily take it out.

File: ops/model.py (hunk 1 above)
Comment (jameinel) [reply ops/model.py:54]: I'm happy to make model_name mandatory keyword, I don't think it is worth changing the signature for the existing parameters.

Comment (chipaca) [reply ops/model.py]: yeah, that's what i meant (i realise how it reads the other way tho)

File: ops/main.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (chipaca) [reply ops/main.py]: in what cases wouldn't it be set? is there a default value we could give it that unambiguously says "was not set" without it potentially breaking charms that expect it to be a string?

**Comment by jameinel** [comment]
Sorry I missed this second comment.
Mostly it isn't set inside our own test suite.
It might not be set by certain versions of Juju (I think that is pre 2.7
but I don't know for sure).
An empty string is pretty clearly not set but not a valid model name. I
felt None was easier to convey a problem then ''. As the latter will 'work'
but not actually do what you want.

**Pull request merged**